data/validation/constraints.ts

```diff
@@ -27,19 +27,65 @@ export const required: Constraint = ({value, displayName}) => {
 };
 
 /**
- * Validate an email address.
- * https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript/46181#46181.
+ * Validate a single email address in a field that expects only one email address.
  */
 export const validEmail: Constraint<string> = ({value, displayName}) => {
     if (isNil(value)) return null;
 
-    // prettier-ignore
-    // eslint-disable-next-line no-useless-escape
-    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
-        isValid = re.test(value);
-    if (!isValid) return `${displayName} is not a properly formatted address.`;
+    const isValid = emailRegEx.test(value);
+    if (!isValid) return `${displayName} is not a properly formatted email address.`;
 };
 
+/**
+ * Validate all email addresses in a field that allows multiple email addresses
+ * separated by a semicolon - separator used by Outlook for multiple email addresses.
+ * All email addresses must be unique.
+ */
+export function validEmails(c?: ValidEmailsOptions): Constraint<string> {
+    return ({value, displayName}) => {
+        if (isNil(value)) return null;
+
+        let emails = value.split(';');
+        if (c?.strict) {
+            if (emails.some(it => it.match(/(^\s|\s$)/)))
+                return `${displayName} must not contain whitespace between email addresses.`;
+            if (value.endsWith(';')) return `${displayName} field must not end with a semicolon.`;
+        }
+
+        emails = emails.filter(Boolean).map(it => it.trim());
+
+        if (uniq(emails).length !== emails.length) {
+            return `${displayName} must not contain duplicate email addresses.`;
+        }
+
+        if (emails.length < c?.minCount) {
+            return `${displayName} must contain at least ${c.minCount} email addresses.`;
+        }
+        if (!isNil(c?.maxCount) && emails.length > c.maxCount) {
+            return `${displayName} must contain no more than ${c.maxCount} email addresses.`;
+        }
+
+        const isValid = emails.every(it => emailRegEx.test(it));
+
+        if (!isValid) return `${displayName} has an improperly formatted email address.`;
+    };
+}
+export interface ValidEmailsOptions {
+    /**
+     * True to not allow whitespace between addresses and not allow a trailing semicolon.
+     * Defaults to false.
+     * False is the more user-friendly option, the user can enter spaces after semicolons, and a trailing semicolon
+     * and the user will not be bothered with a validation error, but downstream code will need to handle trimming.
+     */
+    strict?: boolean;
```

Comment:
I think these options are overkill.  Not hard for app to do the split/count if it needs to validate the number.  And strict can just be fixed at false.

Reply:
I'm ok with removing the `strict`, and hardcoding the `strict:false` code, but can we keep the `min/maxCount`?  Wouldn't it be annoying to a user if their form looked valid and then when they hit submit they got a second surprise message saying they have an error in their email field?  The situation erodes confidence.
Also OK with removing `minCount`, and keeping `maxCount`, but setting `minCount=1` would eliminate the need to import the `required` rule.